socks5.py

```diff
@@ -25,6 +43,62 @@
 except ImportError:
     pass
 
+def is_globally_routable(ipv6_address):
+    non_routable_networks = [
+        "ff00::/8",         # Multicast address range
+        "fe80::/10",        # Link-local address range
+        "fc00::/7",         # Unique local address range
+        "::/8",             # Unspecified address range
+        "2001:db8::/32",    # Documentation address range
+        "2001::/32",        # Teredo address range
+        "2002::/16",        # 6to4 address range
+        "ff02::/16",        # Link-local multicast address range
+    ]
+    for network in non_routable_networks:
+        if ipaddress.ip_address(ipv6_address) in ipaddress.ip_network(network):
+            return False
+    return True
+
+try:
+    # TODO: configurable DNS (or find a way to use the cell network's own DNS)
```

Comment:
I know this is not modified from the original versions, but I'd like to know if you have any ideas on fixing the DNS limitation here. First, a hardcoded list of DNS servers is not ideal, and second, this will probably break on IPv6-only networks.